File: aat-lib/src/main/java/ch/bailu/aat_lib/map/Point.java
package ch.bailu.aat_lib.map;

public class Point {

    public int x, y;

    public Point(int x2, int y2) {
        x = x2;
        y = y2;
    }

    public Point() {
        x=y=0;
    }

    public Point(float x2, float y2) {
        x = Math.round(x2);
        y = Math.round(y2);
    }

    public Point(double x2, double y2) {
        x = Math.round((float)x2);
        y = Math.round((float)y2);
    }

    public void set(Point p) {
        x = p.x;
        y = p.y;
    }

    @Override
    public String toString() {
        return "Point{" +
                "x=" + x +
                ", y=" + y +
                '}';
    }
}

File: aat-lib/src/main/java/ch/bailu/aat_lib/util/Point.kt
package ch.bailu.aat_lib.util

import kotlin.math.roundToInt

/**
 * A point holding two integer coordinates.
 * To keep some code independent from UI libraries.
 */
class Point {
    @JvmField
    var x: Int
    @JvmField
    var y: Int

    constructor(x: Int, y: Int) {
        this.x = x
        this.y = y
    }

    constructor() {
        y = 0
        x = y
    }

    constructor(x: Float, y: Float) {
        this.x = x.roundToInt()
        this.y = y.roundToInt()
    }

    constructor(x: Double, y: Double) {
        this.x = x.toFloat().roundToInt()
        this.y = y.toFloat().roundToInt()
    }

    fun set(p: Point) {
        x = p.x
        y = p.y
    }

    override fun toString(): String {
        return "Point{" +
                "x=" + x +
                ", y=" + y +
                '}'
    }
}
